### ISSUE TITLE
error in DIANNtoMSstatsFormat 'Error: column(s) not found: [FragmentQuantCorrected]'

### ISSUE THREAD

**Issue by bnewt2k** [vdemichev/DiaNN#1088]
Hi All,
Using recently dl/installed DIANN 1.9

I am trying to convert a DIANN Run to MSStats format and I get this error.
 'Error: column(s) not found: [FragmentQuantCorrected]'

I opened report.tsv and I could not find column FragmentQuantCorrected.  Closest column I find is Fragment.Quant.Raw.
Did new version remove FragmentQuantCorrected?  Do you think I could use Fragment.Quant.Raw as  FragmentQuantCorrected?

Thanks,
B

**Comment by vdemichev** [comment]
Hi B,

Yes, can just rename the column. The reason the corrected one was removed is because the correction algorithm was removed from DIA-NN, as it’s not needed with QuantUMS.

Best,
Vadim

**Comment by bnewt2k** [comment]
Hi Vadim,

Thanks for the reply and explanation.  I did modify the report.tsv in Excel by duplicating  Fragment.Quant.Raw and renaming one Fragment.Quant.Corrected, and then the converter ran.

Thanks,
B

**Comment by ABanaeiEsfahani** [comment]
Hi Vadim,

I have a similar question and got the same error. But it seems Fragment.Quant.Raw is not there as well. Any suggestion is highly appreciated.

Best wishes,
Amir


> colnames(df)
  [1] "Run.Index"                    "Run"                          "Channel"                      "Precursor.Id"                
  [5] "Modified.Sequence"            "Stripped.Sequence"            "Precursor.Charge"             "Precursor.Lib.Index"         
  [9] "Decoy"                        "Proteotypic"                  "Precursor.Mz"                 "Protein.Ids"                 
 [13] "Protein.Group"                "Protein.Names"                "Genes"                        "RT"                          
 [17] "iRT"                          "Predicted.RT"                 "Predicted.iRT"                "IM"                          
 [21] "iIM"                          "Predicted.IM"                 "Predicted.iIM"                "Precursor.Quantity"          
 [25] "Precursor.Normalised"         "Ms1.Area"                     "Ms1.Normalised"               "Ms1.Apex.Area"               
 [29] "Ms1.Apex.Mz.Delta"            "Normalisation.Factor"         "Quantity.Quality"             "Empirical.Quality"           
 [33] "Normalisation.Noise"          "Ms1.Profile.Corr"             "Evidence"                     "Mass.Evidence"               
 [37] "Channel.Evidence"             "Ms1.Total.Signal.Before"      "Ms1.Total.Signal.After"       "RT.Start"                    
 [41] "RT.Stop"                      "FWHM"                         "PG.TopN"                      "PG.MaxLFQ"                   
 [45] "Genes.TopN"                   "Genes.MaxLFQ"                 "Genes.MaxLFQ.Unique"          "PG.MaxLFQ.Quality"           
 [49] "Genes.MaxLFQ.Quality"         "Genes.MaxLFQ.Unique.Quality"  "Q.Value"                      "PEP"                         
 [53] "Global.Q.Value"               "Lib.Q.Value"                  "Peptidoform.Q.Value"          "Global.Peptidoform.Q.Value"  
 [57] "Lib.Peptidoform.Q.Value"      "PTM.Site.Confidence"          "Site.Occupancy.Probabilities" "Protein.Sites"               
 [61] "Lib.PTM.Site.Confidence"      "Translated.Q.Value"           "Channel.Q.Value"              "PG.Q.Value"                  
 [65] "PG.PEP"                       "GG.Q.Value"                   "Protein.Q.Value"              "Global.PG.Q.Value"           
 [69] "Lib.PG.Q.Value"               "Best.Fr.Mz"                   "Best.Fr.Mz.Delta"             "Ms2.Scan"                    
 [73] "Fr.0.Index"                   "Fr.0.Id"                      "Fr.0.Quantity"                "Fr.0.Score"                  
 [77] "Fr.1.Index"                   "Fr.1.Id"                      "Fr.1.Quantity"                "Fr.1.Score"                  
 [81] "Fr.2.Index"                   "Fr.2.Id"                      "Fr.2.Quantity"                "Fr.2.Score"                  
 [85] "Fr.3.Index"                   "Fr.3.Id"                      "Fr.3.Quantity"                "Fr.3.Score"                  
 [89] "Fr.4.Index"                   "Fr.4.Id"                      "Fr.4.Quantity"                "Fr.4.Score"                  
 [93] "Fr.5.Index"                   "Fr.5.Id"                      "Fr.5.Quantity"                "Fr.5.Score"                  
 [97] "Fr.6.Index"                   "Fr.6.Id"                      "Fr.6.Quantity"                "Fr.6.Score"                  
[101] "Fr.7.Index"                   "Fr.7.Id"                      "Fr.7.Quantity"                "Fr.7.Score"                  
[105] "Fr.8.Index"                   "Fr.8.Id"                      "Fr.8.Quantity"                "Fr.8.Score"                  
[109] "Fr.9.Index"                   "Fr.9.Id"                      "Fr.9.Quantity"                "Fr.9.Score"                  
[113] "Fr.10.Index"                  "Fr.10.Id"                     "Fr.10.Quantity"               "Fr.10.Score"                 
[117] "Fr.11.Index"                  "Fr.11.Id"                     "Fr.11.Quantity"               "Fr.11.Score"

**Comment by vdemichev** [comment]
Hi Amir,

The output format changed in DIA-NN since 1.9-series. Are you using the latest MSStats?

Best,
Vadim

**Comment by ABanaeiEsfahani** [comment]
Yes, my MSstats version is 4.14.2! It's good to have some documents on the column names. Some are not self explanatory!

Best wishes,
Amir

**Comment by vdemichev** [comment]
Fragment.Quant.Raw is just no longer produced. Please see docs on column names primarily here https://github.com/vdemichev/DiaNN?tab=readme-ov-file#output but also elsewhere in the user guide.

**Comment by ABanaeiEsfahani** [comment]
So there is not fragment quantity in the output anymore? what should I use instead?

**Comment by ABanaeiEsfahani** [comment]
It seems that the --export-quant flag does not add fragment quant to the report anymore! I would really appreciate your advice! from which version of DIA-NN they are not reported anymore? I currently use the latest version (2.3.2) with the following args!

args <- c(
  "--threads", "64",
  "--qvalue", "0.01",
  "--matrices",
  "--export-quant",
  "--predictor",
  "--met-excision",
  "--fasta-search",
  "--fasta", "/scicore/home/gagneux/GROUP/proteomics/TbX021/Tools/uniprotkb_proteome_UP000001584_2026_02_19.fasta",
  "--full-profiling",
  "--cut", "K*,R*,!*P",
  "--missed-cleavages", "1",
  "--min-pep-len", "6",
  "--max-pep-len", "30",
  "--min-pr-charge", "2",
  "--max-pr-charge", "4",
  "--min-pr-mz", "350",
  "--max-pr-mz", "1500",
  "--min-fr-mz", "120",
  "--max-fr-mz", "900",
  "--fixed-mod", "UniMod:4,57.021464,C",
  "--var-mod", "UniMod:35,15.994915,M",
  "--var-mod", "UniMod:21,79.966331,STY",
  "--var-mod", "UniMod:1,42.010565,*n",
  "--var-mods", "3",
  "--peptidoforms",
  "--site-ms1-quant",
  "--mass-acc", "15",
  "--mass-acc-ms1", "10",
  "--individual-mass-acc",
  "--individual-windows",
  "--reanalyse",
  "--quant-sel-runs", "9",
  "--matrix-qvalue", "0.01",
  "--verbose", "3",
  "--dir", "/scicore/home/gagneux/GROUP/proteomics/TbX021/RawData",
  "--out", "/scicore/home/gagneux/GROUP/proteomics/TbX021/Results/DIANN2/Main_Report.tsv",
  "--out-lib", "/scicore/home/gagneux/GROUP/proteomics/TbX021/Results/DIANN2/Lib_Report.tsv",
  "--export-quant",
  "--temp", "/scicore/home/gagneux/GROUP/proteomics/TbX021/Results/DIANN2/tmp"
)

**Comment by vdemichev** [comment]
Hi Amir,

Please configure and test under Windows, and then copy commands it generates to Linux, otherwise easy to configure incorrectly. There are multiple issues with the settings above (I assume they were copied form old settings used for 1.8.1).

Best,
Vadim

**Comment by vdemichev** [comment]
> So there is not fragment quantity in the output anymore? what should I use instead?

[73] "Fr.0.Index" "Fr.0.Id" "Fr.0.Quantity" "Fr.0.Score"
[77] "Fr.1.Index" "Fr.1.Id" "Fr.1.Quantity" "Fr.1.Score"
[81] "Fr.2.Index" "Fr.2.Id" "Fr.2.Quantity" "Fr.2.Score"
[85] "Fr.3.Index" "Fr.3.Id" "Fr.3.Quantity" "Fr.3.Score"
[89] "Fr.4.Index" "Fr.4.Id" "Fr.4.Quantity" "Fr.4.Score"
[93] "Fr.5.Index" "Fr.5.Id" "Fr.5.Quantity" "Fr.5.Score"
[97] "Fr.6.Index" "Fr.6.Id" "Fr.6.Quantity" "Fr.6.Score"
[101] "Fr.7.Index" "Fr.7.Id" "Fr.7.Quantity" "Fr.7.Score"
[105] "Fr.8.Index" "Fr.8.Id" "Fr.8.Quantity" "Fr.8.Score"
[109] "Fr.9.Index" "Fr.9.Id" "Fr.9.Quantity" "Fr.9.Score"
[113] "Fr.10.Index" "Fr.10.Id" "Fr.10.Quantity" "Fr.10.Score"
[117] "Fr.11.Index" "Fr.11.Id" "Fr.11.Quantity" "Fr.11.Score"

**Comment by ABanaeiEsfahani** [comment]
Hi Vadim,

Thanks for the comments. I just configured it using the GUI on Windows. It seems that the args are very similar! which parts do you see problematic? I needed to remove --individual-mass-acc and  --individual-windows, right?

Concerning the MSstats workflow and fragments ions, I am trying to reformat them. But I just realized that Precursor.Quantity which is supposed to be the sum of top3 fragment ions, is much larger usually even when I sum all fragment ions not only top3? Could you please help me understand what I missed?

Best wishes,
Amir

**Comment by vdemichev** [comment]
Hi Amir,

output in .tsv --fasta-search and --predictor combined with raw data analysis, very excessive --var-mod config, --full-profiling, --export-quant passed twice, mass accuracies and scan window not fixed, despite this being a quantitative analysis (otherwise why all those settings).

Best,
Vadim

**Comment by ABanaeiEsfahani** [comment]
Hi Vadim,

Thanks for the comments. I will implement them in. very helpful. Would you comment on the fragment ion issue as well?

Best wishes,
Amir

**Comment by vdemichev** [comment]
> But I just realized that Precursor.Quantity which is supposed to be the sum of top3 fragment ions

Not exactly, also not with QuantUMS. Please see the current version of the user guide.

**Comment by vdemichev** [comment]
>  I am trying to reformat them

MSStats team confirmed the 2.0+ DIA-NN not supported? I somehow thought it was. Why use MSStats in the first place then?

**Comment by ABanaeiEsfahani** [comment]
> > I am trying to reformat them
> 
> MSStats team confirmed the 2.0+ DIA-NN not supported? I somehow thought it was. Why use MSStats in the first place then?

It seems not really. I get the following error. What would you recommend for DE analysis? I have used it for a long time and continued running this. But I would be happy to take another one!

INFO  [2026-02-26 14:46:28] ** Raw data from DIANN imported successfully.
Error in `[.data.table`(dn_input, , req_cols, with = FALSE) : 
  column(s) not found: [FragmentQuantCorrected

**Comment by vdemichev** [comment]
> What would you recommend for DE analysis?

limma, it's really excellent, easy to use (with AI completely trivial) and methods used are very well thought out

**Comment by ABanaeiEsfahani** [comment]
> > What would you recommend for DE analysis?
> 
> limma, it's really excellent, easy to use (with AI completely trivial) and methods used are very well thought out

And you run it on precursor quantities computed by QuantUMS, right? not fragment ions?
BTW, QuantUMS paper is not out yet? I mean the final version not bioRxiv one!

**Comment by vdemichev** [comment]
Directly on protein quantities ('MaxLFQ' ones), but can also run on precursor quantities (Precursor.Normalised).

> BTW, QuantUMS paper is not out yet? I mean the final version not bioRxiv one!

All in due time :)

**Comment by ABanaeiEsfahani** [comment]
> Directly on protein quantities ('MaxLFQ' ones), but can also run on precursor quantities (Precursor.Normalised).
> 
> > BTW, QuantUMS paper is not out yet? I mean the final version not bioRxiv one!
> 
> All in due time :)

OK, all sounds good. My last Q! If I still want to run MSstats, do you foresee any issue if I run it on precursor quantities (I guess unnormalized because MSstats itself does the job!, right?), not fragment ions?

**Comment by vdemichev** [comment]
I am not sure, please ask MSStats support about this. In general, I assume the goal of MSStats is to perform analysis as well as possible based on raw data produced by the search engine. This makes sense for many search engine, however DIA-NN since its inception has been highly optimised towards achieving good quantification, so what MSStats is doing may be either partially redundant with what DIA-NN does or may not be able to reproduce it (e.g. to achieve sth comparable to QuantUMS need to basically reproduce the algorithm or implement something comparably sophisticated).
